canal/sync.go

```diff
@@ -16,7 +16,7 @@ import (
 
 func (c *Canal) startSyncer() (*replication.BinlogStreamer, error) {
 	gset := c.master.GTIDSet()
-	if gset == nil {
+	if gset == nil || (gset != nil && gset.String() == "") {
```

Comment:
```
(gset != nil && gset.String() == "")
```
seem here we can use `gset.String()` directly beause the first condition has already checked gset is nil.

Reply:
Yes